### DISCUSSION TITLE
现在Dva已经停止维护了，请问下学习umi是否有必要先学习dva的核心知识？

### DISCUSSION THREAD

**Discussion by liaodalin19903** [umijs/umi#10418] [Q&A] [answered]
现在Dva已经停止维护了，请问下学习umi是否有必要先学习dva的核心知识？

**Comment by fz6m** [comment, marked as answer]
umi 中所有功能都是插件化的，任何附加功能都是独立的插件，可以选择性开启使用，有需求再开启使用。

比如 `antd` 插件，当需要使用时再开启，`dva` 同理，现在还存在 `dva` 主要是 umi 3 升级和历史遗留。

新开发项目无需再学习，建议使用更现代的数据方案（zustand 、jotai 、 react-query 、 valtio 等）

**Reply by yuchengWu630** [reply]
那官方文档拜托更新一下，我现在用的4.0.7开发到一半才发现dva被弃用，根本没办法在外部的js中修改models的值，等于要全局替换jotai 、 valtio代价太大

**Comment by xiaohuoni** [comment]
我个人意见还是可以学习一下 dva 的，感觉概念设计的非常适合新手上手数据流。
